Sources/RegexDSL/DSL.swift

```diff
@@ -23,7 +23,10 @@ extension CharacterClass: RegexProtocol {
   public typealias Capture = Empty
 
   public var regex: Regex<Capture> {
-    .init(ast: .characterClass(self))
+    guard let ast = self.makeAST() else {
+      fatalError("FIXME: extended AST?")
```

Comment:
@rxwei this works for prototyping, but at some point we want a result builder that can run custom code. Have you had thoughts about how this would work? Extend the AST, or an extended AST, or something else entirely?

Reply:
I think the builder DSL will have it's own AST, with embedded Regex AST as a case when applicable.